virtual_ship/expedition/do_expedition.py

```diff
@@ -0,0 +1,149 @@
+from .ship_config import ShipConfig
+from .simulate_measurements import simulate_measurements
+from .simulate_schedule import ScheduleProblem, simulate_schedule
+from .verify_schedule import verify_schedule
+
+
+def do_expedition(expedition_dir: str | Path) -> None:
+    """
+    Perform an expedition, providing terminal feedback and file output.
+
+    :param expedition_dir: The base directory for the expedition.
+    """
+    if isinstance(expedition_dir, str):
+        expedition_dir = Path(expedition_dir)
+
+    # load ship configuration
+    ship_config = _get_ship_config(expedition_dir)
+    if ship_config is None:
+        return
+
+    # load schedule
+    schedule = _get_schedule(expedition_dir)
+    if schedule is None:
+        return
+
+    # load last checkpoint
+    checkpoint = _load_checkpoint(expedition_dir)
+    if checkpoint is None:
+        checkpoint = Checkpoint(past_schedule=Schedule(waypoints=[]))
+
+    # verify that schedule and checkpoint match
+    if (
+        not schedule.waypoints[: len(checkpoint.past_schedule.waypoints)]
+        == checkpoint.past_schedule.waypoints
+    ):
+        print(
+            "Past waypoints in schedule have been changed! Restore past schedule and only change future waypoints."
+        )
+        return
+
+    # projection used to sail between waypoints
+    projection = pyproj.Geod(ellps="WGS84")
+
+    # load fieldsets
+    input_data = _load_input_data(
+        expedition_dir=expedition_dir, ship_config=ship_config
+    )
+
+    # verify schedule makes sense
+    verify_schedule(projection, ship_config, schedule, input_data)
+
+    # simulate the schedule
+    schedule_results = simulate_schedule(
+        projection=projection, ship_config=ship_config, schedule=schedule
+    )
+    if isinstance(schedule_results, ScheduleProblem):
+        print(
+            "Update your schedule and continue the expedition by running the tool again."
+        )
+        _save_checkpoint(
+            Checkpoint(
+                past_schedule=Schedule(
+                    waypoints=schedule.waypoints[: schedule_results.failed_waypoint_i]
+                )
+            ),
+            expedition_dir,
+        )
+        return
+
+    # delete and create results directory
+    if os.path.exists(expedition_dir.joinpath("results")):
+        shutil.rmtree(expedition_dir.joinpath("results"))
+    os.makedirs(expedition_dir.joinpath("results"))
+
+    # calculate expedition cost in US$
+    assert (
+        schedule.waypoints[0].time is not None
+    ), "First waypoint has no time. This should not be possible as it should have been verified before."
+    time_past = schedule_results.time - schedule.waypoints[0].time
+    cost = expedition_cost(schedule_results, time_past)
+    with open(expedition_dir.joinpath("results", "cost.txt"), "w") as file:
+        file.writelines(f"cost: {cost} US$")
+    print(f"This expedition took {time_past} and would have cost {cost:,.0f} US$.")
```

Comment:
Would it not make more sense to display this _after_ the simulate_measurements? Or at least also print it then?